library/src/main/java/com/novoda/downloadmanager/lib/DownloadThread.java

```diff
@@ -845,14 +863,40 @@ private void addRequestHeaders(State state, HttpURLConnection conn) {
     private void notifyDownloadCompleted(State state, int finalStatus, String errorMsg, int numFailed) {
         notifyThroughDatabase(state, finalStatus, errorMsg, numFailed);
         if (DownloadStatus.isCompleted(finalStatus)) {
-            originalDownloadInfo.broadcastIntentDownloadComplete(finalStatus);
+            broadcastIntentDownloadComplete(finalStatus);
         } else if (DownloadStatus.isInsufficientSpace(finalStatus)) {
-            originalDownloadInfo.broadcastIntentDownloadFailedInsufficientSpace();
+            broadcastIntentDownloadFailedInsufficientSpace();
         }
     }
 
+    public void broadcastIntentDownloadComplete(int finalStatus) {
+        Intent intent = new Intent(DownloadManager.ACTION_DOWNLOAD_COMPLETE);
+        intent.setPackage(getPackageName());
+        intent.putExtra(DownloadManager.EXTRA_DOWNLOAD_ID, originalDownloadInfo.getId());
+        intent.putExtra(DownloadManager.EXTRA_DOWNLOAD_STATUS, finalStatus);
+        intent.setData(originalDownloadInfo.getMyDownloadsUri());
+        if (originalDownloadInfo.getExtras() != null) {
```

Comment:
cc @xrigau we should rename this method `originalDownloadInfo.getNotificationExtras()`